frontend/src/Frontend/UI/Transfer.hs

```diff
@@ -1280,6 +1330,8 @@ getKeysetActionSingle c (Just a) mDetails =
     Just (AccountStatus_Exists (AccountDetails _ g)) -> do
       case g of
         AccountGuard_Other _ -> KeysetNoAction
+        -- TODO: needs better comment below lol
+        AccountGuard_KeySetRef refName -> KeysetNoAction -- this should be resolved at an earlier step
```

Comment:
Here is a great example of where the above `KeySetHeritage` type and `KeySetLike` constructor would be useful.  The comment here says that the keyset-ref should be resolved at an earlier step, but it's not immediately obvious from a quick review whether that is actually true or not.  It's possible that doing `KeysetNoAction` here covers up some important cases.  But if you had `KeySetLike ksh`, then all the needed information that should have been resolved earlier would definitely be included.

Reply:
I agree. Since we now use `KeySetHeritage`, we either hit the `AccountGuard_Other` case (which means that keyset-ref has not been resolved, which is possible, if the keyset-ref is invalid) or we hit the case `AccountGuard_KeySetLike` which includes the `KeySetHeritage` in its constructor, which could contain a keyset-ref along with its accompanying keyset.